e2e/cypress/integration/api/iam/04_teams_api.spec.ts

```diff
@@ -0,0 +1,329 @@
+            method: 'POST',
+            failOnStatusCode: false,
+            body: teamWithProjects(teamID, [project1.id, project2.id])
+        }).then((response) => {
+            expect(response.status).to.equal(403);
+        });
+      });
+    });
+
+    describe('PUT /apis/iam/v2beta/teams', () => {
+      it('admin can update a team with no projects to have projects', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+
+        cy.request({ ...defaultAdminReq,
+            method: 'PUT',
+            url: `/apis/iam/v2beta/teams/${teamID}`,
+            body: teamWithProjects(teamID, [project1.id, project2.id])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(2);
+        });
+      });
+
+      it('admin can update a team with projects to have no projects', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [project1.id, project2.id])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(2);
+        });
+
+        cy.request({ ...defaultAdminReq,
+            method: 'PUT',
+            url: `/apis/iam/v2beta/teams/${teamID}`,
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+      });
+
+      it('admin cannot update a team to have projects that do not exist', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+
+        cy.request({ ...defaultAdminReq,
+            method: 'PUT',
+            url: `/apis/iam/v2beta/teams/${teamID}`,
+            failOnStatusCode: false,
+            body: teamWithProjects(teamID, [project1.id, 'notfound'])
+        }).then((response) => {
+            expect(response.status).to.equal(404);
+        });
+      });
+
+      it('non-admin with project1 assignment access can update ' +
+      'a team with no projects to have project1', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+
+        cy.request({ ...defaultNonAdminReq,
+            method: 'PUT',
+            url: `/apis/iam/v2beta/teams/${teamID}`,
+            body: teamWithProjects(teamID, [project1.id])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(1);
+        });
+      });
+
+      it('non-admin with project1 assignment access can update ' +
```

Comment:
Is it possible to pull these tests into two context blocks, one with the non-admin and one with the admin user? All the tests are duped so if we could just loop through those scenarios, that would be 💯